### PR TITLE
POC: explore running plugin as api-server

### PR DESCRIPTION
**Pull request by ryantxu**
early early early early

but lets consider what it would look like to run a plugin as an agregated API server

Without k8s:
```
18M	dist/gpx_sheets_darwin_arm64
```

With k8s:
```
69M	dist/gpx_sheets_darwin_arm64
```
so ~50mb to include the api server (so far)

### CODE REVIEW

File: pkg/ext/server.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	stream io.ReadCloser, flush bool, mimeType string, err error)
+
+// This is used to implement dynamic sub-resources like pods/x/logs
+type RawAPIHandler struct {
+	Path    string
+	OpenAPI string
+	Level   RawAPILevel // resource | namespace | group
+
+	// The GET request + response (see the standard /history and /refs)
+	Handler func(ctx context.Context, id kindsys.StaticMetadata) (http.HandlerFunc, error)
```
Comment (charandas) [inline pkg/ext/server.go:49]: Used to be `Handler func(ctx context.Context, id StaticMetadata) StreamingResponse`. I have gone ahead and converted the usage to use a plain `http.HandlerFunc`.  Having an `error` helps establish some certainty as to why in caller code if the Handler func that is supposed to return a HandlerFunc fails.

**Comment by gabor** [comment]
hi @ryantxu & @charandas , could you add some more info about what we are trying to achieve here, please? i mean, i guess this is something kubernetes related, but what exactly? thanks!

**Comment by ryantxu** [comment]
@gabor -- sorry for the delay!  This is not intended for merge -- really an exploration of how to expose plugins in k8s.  Using a real datasource makes this more concrete, but the PR/changeset is just for explortation :)

**Comment by gabor** [comment]
> @gabor -- sorry for the delay! This is not intended for merge -- really an exploration of how to expose plugins in k8s. Using a real datasource makes this more concrete, but the PR/changeset is just for explortation :)

thanks for the info. sounds fun :).... is this thing able to run? any quick notes on how to run it maybe? (i know, i know, this is just exploration, but it's hard to resist)

**Comment by ryantxu** [comment]
> any quick notes on how to run it maybe? 

We will aim to get a testable version without complex dependency setup working next week 🤞🏻

**Comment by ryantxu** [comment]
@gabor -- FYI, this is all in main now so I'm closing here.  Sorry to keep it open so long!

**Pull request closed without merging**